### ISSUE TITLE
help migrate legacy pages/api/ endpoints to tRPC

### ISSUE THREAD

**Issue by PeerRich** [calcom/cal.com#3573]
https://github.com/calcom/cal.com/tree/main/apps/web/pages/api

@zomars wanna make a list of endpoints?

@zomars EDIT:

- [x] `/api/availability/[user]`
  - [x] GET -> query `viewer.availability.user` done in #3591 
- [ ] `/api/availability/calendar`
  - [ ] GET -> query `viewer.selectedCalendars.list`
  - [ ] DELETE -> mutation `viewer.selectedCalendars.delete`
  - [ ] POST -> mutation `viewer.selectedCalendars.createOrUpdate` OR `viewer.availability.calendar.upsert`
- [x] `/api/availability/day` (DEPRECATED)
- [x] `/api/availability/eventtype`
  - [x] POST -> mutation `viewer.eventTypes.create`
  - [x] PATCH -> mutation `viewer.eventTypes.update`
  - [x] DELETE -> mutation `viewer.eventTypes.delete`
- [x] `/api/book/confirm`
  - [ ] GET ->  query `viewer.book.confirm`
- [ ] `/api/book/event`
  - [ ] POST -> mutation `viewer.book.event`
- [ ] `/api/book/request-reschedule`
  - [ ] POST -> mutation `viewer.book.requestReschedule`
- [x] `/api/event-type/index`
  - [x] GET -> query `viewer.eventTypes.list`

WIP...

**Comment by hussamkhatib** [comment]
Hi, I would like to pick this issue

**Comment by PeerRich** [comment]
> Hi, I would like to pick this issue

assigned

**Comment by zomars** [comment]
- [ ] GET: `/api/availability/[user]`
  - [ ] query `viewer.availability.user`
- [ ] `/api/availability/calendar`
  - [ ] GET -> query `viewer.selectedCalendars.list`
  - [ ] DELETE -> mutation `viewer.selectedCalendars.delete`
  - [ ] POST -> mutation `viewer.selectedCalendars.createOrUpdate` OR `viewer.availability.calendar.upsert`
- [ ] `/api/availability/day` (DEPRECATED)
- [ ] `/api/availability/eventtype`
  - [ ] POST -> mutation `viewer.eventTypes.create`
  - [ ] PATCH -> mutation `viewer.eventTypes.update`
  - [ ] DELETE -> mutation `viewer.eventTypes.delete`
- [ ] `/api/book/confirm`
  - [ ] GET ->  query `viewer.book.confirm`
- [ ] `/api/book/event`
  - [ ] POST -> mutation `viewer.book.event`
- [ ] `/api/book/request-reschedule`
  - [ ] POST -> mutation `viewer.book.requestReschedule`
- [ ] `/api/event-type/index`
  - [ ] GET -> query `viewer.eventTypes.list`

WIP...

**Comment by hussamkhatib** [comment]
Hi Maintainers, I am working on this endpoint currently.
> ` /api/book/confirm`
 GET -> `query viewer.book.confirm`

 The above should be a POST method in trpc.
 See below : [BookingListItem.tsx](https://github.com/calcom/cal.com/blob/main/apps/web/components/booking/BookingListItem.tsx)
 ```ts
       const res = await fetch("/api/book/confirm", {
        method: "PATCH",
        body: JSON.stringify(body),
        headers: {
          "Content-Type": "application/json",
        },
      });
 
 ```
 
 
 Also  for `query viewer.book.confirm` should it not be `viewer.booking.confirm` as I can think that is what it is meant for and there already exist a route for that. [viewer/bookings.tsx](https://github.com/calcom/cal.com/blob/main/packages/trpc/server/routers/viewer/bookings.tsx)

**Comment by hussamkhatib** [comment]
The route `/api/event-type/index` route does not exist.

**Comment by hussamkhatib** [comment]
HI @PeerRich , can you update the todos, the below endpoints are completed in previous PR. 
- [x] /api/book/confirm 
- [x] /api/availability/day
- [x] /api/availability/eventtype   // all methods
- [x] /api/event-type

**Comment by zomars** [comment]
Done!

**Comment by hussamkhatib** [comment]
Hey @zomars,
Could help me get through this PR https://github.com/calcom/cal.com/pull/7234
It's WIP and currently blocked by typeScript issue which I am not able to debug it.

**Comment by PeerRich** [comment]
closing due to stale. feel free to open a new issue
